### PR TITLE
MySQL: Add RETURN Statement

### LINKED ISSUE #4648: MySQL: Found unparsable section for "RETURN" and "CREATE VIEW"
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Hey, I would like to your help and thanks for this repo :)

sqlfluff version: 2.0.2

```
CREATE VIEW view_example_list AS
    (
        SELECT name_value
        FROM test_table
        ORDER BY name_value DESC
    );
```

Will raise:

L:  36 | P:   1 |  PRS | Line 36, Position 1: Found unparsable section: 'CREATE
                       | VIEW view_example_list AS\n    (\n ...

```
CREATE FUNCTION test_example_scalar_function()
    RETURNS INT
    DETERMINISTIC
BEGIN
    DECLARE excepted_result INT DEFAULT 1;
    DECLARE actual_result INT;
    DECLARE test_result INT;

    SELECT example_scalar_function('test1') INTO actual_result;

    SET test_result = IF((actual_result != excepted_result OR actual_result IS NULL), 0, 1);

    RETURN test_result;
END;
```

Will raise:

L:  30 | P:   5 |  PRS | Line 30, Position 5: Found unparsable section: 'RETURN
                       | result

### Expected Behaviour

Will not raise any error.

### Observed Behaviour

 Found unparsable section

### How to reproduce

in the section above.

### Dialect

MySql

### Version

2.0.2

### Configuration

```
[sqlfluff]
dialect = mysql
max_line_length = 100
exclude_rules = L003, L030, L036, L039, L044

[sqlfluff.core]
sql_file_exts = .sql

[sqlfluff.indentation]
tab_space_size = 4
indent_unit = space

[sqlfluff:rules]
allow_scalar = True

[sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by yoichi**
### Brief summary of the change made

fixes #4648 

### Are there any other side effects of this change that we should be aware of?

No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -2646,3 +2647,16 @@ class AlterOptionSegment(BaseSegment):
...
+class ReturnStatementSegment(BaseSegment):
+    """A RETURN statement.
+
+    As specified in https://dev.mysql.com/doc/refman/8.0/en/return.html
+    """
+
+    type = "return_statement"
+    match_grammar = Sequence(
+        "RETURN",
+        Ref("ExpressionSegment", optional=True),
```
Comment (yoichi) [inline src/sqlfluff/dialects/dialect_mysql.py:2661]: optional=True is unnecessary

Comment (yoichi) [reply src/sqlfluff/dialects/dialect_mysql.py]: fixed in 6a8f061f

Comment (yoichi) [reply src/sqlfluff/dialects/dialect_mysql.py]: > optional=True is unnecessary

fixed in 6a8f061f

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**